server.js

```diff
@@ -0,0 +1,41 @@
+// server.js
+const { createServer } = require('http')
+const { parse } = require('url')
+const next = require('next')
+
+const dev = process.env.NODE_ENV !== 'production'
+const app = next({ dev })
+const handle = app.getRequestHandler()
+
+// this is the logger for the server
+var logger = require('pino-http')()
+
+app.prepare().then(() => {
+  createServer((req, res) => {
+    // Be sure to pass `true` as the second argument to `url.parse`.
+    // This tells it to parse the query portion of the URL.
+    const parsedUrl = parse(req.url, true)
+    const { pathname, query } = parsedUrl
+    logger(req, res)
+
+    if (pathname === '/log') {
+        let data = null
+        req.on('data', chunk => {
+            data = chunk
+        })
+        req.on('end', () => {
+            const { msg, level = 'info' } = JSON.parse(data)
+            req.log[level](msg)
+        })
```

Comment:
Please use the `stream.finished()` utility inside end.

Reply:
I've added the ok response inside that block, but I'm not sure how to use `stream.finished()` ?

Reply:
https://nodejs.org/api/stream.html#stream_stream_finished_stream_options_callback

Reply:
done, thanks!